Send the autosize column event for grid native header columns

thesiv opened the pull request:
Send wxEVT_GRID_COL_AUTO_SIZE on double clicking on a separator line of
the grid native header to allow override default behaviour.

thesiv committed:
Send the autosize column event for grid native header columns

Send wxEVT_GRID_COL_AUTO_SIZE on double clicking on a separator line of
the grid native header to allow override default behaviour.

vadz reviewed (commented):
This looks like a sufficient minimal fix, but could we rather extract the existing code from `wxGrid::ProcessColLabelMouseEvent()`, i.e. this part:
```cpp
            // adjust column width depending on label text
            //
            // TODO: generate RESIZING event, see #10754
            if ( !SendGridSizeEvent(wxEVT_GRID_COL_AUTO_SIZE, -1, colEdge, event) )
                AutoSizeColLabelSize( colEdge );

            SendGridSizeEvent(wxEVT_GRID_COL_SIZE, -1, colEdge, event);
```
into some `wxGrid::HandleColumnAutosize()` and call it from both there and here?

This would result in also sending `wxEVT_GRID_COL_SIZE` when using the native header, which seems like a good thing.

thesiv committed:
fixup! Send the autosize column event for grid native header columns

thesiv committed:
Fix wxGrid::SetColSize for the native header

In case of the width of -1 for the native header the title width must be
calculated by the native header itself.

thesiv committed:
Fix the native header column minimal width determination

wxGrid::AutoSizeColumn can set per column minimal width and this values
must be used for native headers columns.
